RULES/hdl/questa.go

```diff
@@ -2,75 +2,486 @@ package hdl
 
 import (
 	"fmt"
+	"path"
+	"strings"
+	"log"
+	"os"
 
 	"dbt-rules/RULES/core"
-	"dbt-rules/hdl"
 )
 
-type QuestaSimScriptParams struct {
-	Name         string
-	PartName     string
-	BoardName    string
-	OutDir       core.Path
-	OutScript    core.Path
-	OutSimScript core.Path
-	IncDir       core.Path
-	Srcs         []core.Path
-	Ips          []core.Path
-	Libs         []string
-	LibDir       string
-	Verbose      bool
+// VlogFlags enables the user to specify additional flags for the 'vlog' command.
+var VlogFlags = core.StringFlag{
+	Name: "questa-vlog-flags",
+	DefaultFn: func() string {
+		return ""
+	},
+}.Register()
+
+// VcomFlags enables the user to specify additional flags for the 'vcom' command.
+var VcomFlags = core.StringFlag{
+	Name: "questa-vcom-flags",
+	DefaultFn: func() string {
+		return ""
+	},
+}.Register()
+
+// VsimFlags enables the user to specify additional flags for the 'vsim' command.
+var VsimFlags = core.StringFlag{
+	Name: "questa-vsim-flags",
+	DefaultFn: func() string {
+		return ""
+	},
+}.Register()
+
+// Access enables the user to control the accessibility in the compiled design for
+// debugging purposes.
+var Access = core.StringFlag{
+	Name: "questa-access",
+	DefaultFn: func() string {
+		return "rna"
+	},
+}.Register()
+
+// Coverage enables the user to run the simulation with code coverage.
+var Coverage = core.BoolFlag{
+	Name: "questa-coverage",
+	DefaultFn: func() bool {
+		return false
+	},
+}.Register()
+
+// Lib returns the standard Questa library name defined for this rule.
+func (rule Simulation) Lib() string {
+	return rule.Name + "Lib"
+}
+
+// Target returns the optimization target name defined for this rule.
+func (rule Simulation) Target() string {
+	if Coverage.Value() {
+		return rule.Name + "Cov"
+	} else {
+		return rule.Name
+	}
+}
+
+// Path returns the default root path for log files defined for this rule.
+func (rule Simulation) Path() core.Path {
+	return core.BuildPath("/" + rule.Name)
+}
+
+// Instance returns the instance name of the rule based on the Top and the DUT
+// fields.
+func (rule Simulation) Instance() string {
+	// Defaults
+	top := "board"
+	dut := "u_dut"
+
+	// Pick top from rule
+	if rule.Top != "" {
+		top = rule.Top
+	} 
+
+	// Pick DUT from rule
+	if rule.Dut != "" {
+		dut = rule.Dut
+	} 
+
+	return "/" + top + "/" + dut
+}
+
+// rules holds a map of all defined rules to prevent defining the same rule
+// multiple times.
+var rules = make(map[string]bool)
+
+// common_flags holds common flags used for the 'vlog', 'vcom', 'vopt' and 'vsim' commands.
+const common_flags = "-nologo -quiet"
```

Comment:
I don't think you're adding the `common_flags` to the vsim command, do you?

Reply:
No, the `vsim` command needs different flags. I fixed the comment.